api/app/controllers/webhook.controller.ts

```diff
@@ -174,109 +178,120 @@ async function processCommit(
-    }
+    try {
+      const rawUrl = `https://api.github.com/repos/${pushEvent.repository?.full_name}/contents/${file}?ref=${branch}`;
+      const response = await axios.get(rawUrl, {
+        headers: { Accept: "application/vnd.github.v3.raw" },
+      });
+      const mdContent = response.data;
+      const jsonContent: BaseParsedMdContent =
+        parseMdToJSON<BaseParsedMdContent>(mdContent);
+      const transcript_by = jsonContent.transcript_by.toLowerCase();
+
+      function isTranscriptValid(transcript_by: string): boolean {
+        const lowerCaseTranscriptBy = transcript_by.toLowerCase();
+        return (
+          lowerCaseTranscriptBy.includes("tstbtc") &&
+          lowerCaseTranscriptBy.includes("--needs-review")
+        );
+      }
 
-    const transcriptHash = generateUniqueHash(jsonContent);
-    const totalWords = getTotalWords(jsonContent.body);
-    const content = jsonContent;
+      if (!jsonContent) {
+        throw new Error(
+          "Malformed data: transcript content might not be in the correct format"
+        );
+      }
 
-    if (!transcriptHash || !totalWords) {
-      throw new Error(
-        "Malformed data: transcript content might not be in the correct format"
-      );
-    }
+      const totalWords = getTotalWords(jsonContent.body);
+      const content = jsonContent;
 
-    const existingTranscript = await Transcript.findOne({
-      where: { transcriptUrl: rawUrl },
-    });
+      if (!totalWords) {
+        throw new Error(
+          "Malformed data: transcript content might not be in the correct format"
+        );
+      }
 
-    if (existingTranscript && type === "added") {
-      throw new Error("transcript already exists");
-    }
+      const existingTranscript = await Transcript.findOne({
+        where: { transcriptUrl: rawUrl },
+      });
 
-    if (!isTranscriptValid(transcript_by)) {
-      throw new Error(
-        "Transcript not from TSTBTC or does not need review - did not queue transcript"
-      );
-    }
+      if (existingTranscript && type === "added") {
+        throw new Error("transcript already exists");
+      }
 
-    const transcript: TranscriptAttributes = {
-      originalContent: {
-        ...content,
-        title: content.title.trim(),
-      },
-      content: content,
-      transcriptHash,
-      transcriptUrl: rawUrl,
-      status: TranscriptStatus.queued,
-      contentTotalWords: totalWords,
-    };
+      if (!isTranscriptValid(transcript_by)) {
+        throw new Error(
+          "Transcript not from TSTBTC or does not need review"
+        );
+      }
 
-    let transcriptData: any;
+      const transcript: TranscriptAttributes = {
+        originalContent: {
+          ...content,
+          title: content.title.trim(),
+        },
+        content: content,
+        transcriptHash: "",
+        transcriptUrl: rawUrl,
+        status: TranscriptStatus.queued,
+        contentTotalWords: totalWords,
+      };
+
+      let transcriptData: any;
```

Comment:
why type `any` and not `TranscriptAttributes`?

Reply:
Quite an intricacy for me here, cos first in the `TranscriptAttributes` interface, transcriptUrl is `string | null`,  wanted to change it to `string | undefined `to be able to use it in the sending alert logic `line 268`. another option is to make transcriptUrl option(don't want to do that, cos its critical for further operation)

Reply:
Not sure I understand this but you can type cast the transcriptUrl to what you want when passing it to the `sendAlert` fn. Ideally, we should have one type for the url